File: back/ers-back/src/main/java/com/revature/rmarinoff/ersback/user/Employee.java
package com.revature.rmarinoff.ersback.user;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;

@Entity
public class Employee {	
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator="employee")
	@SequenceGenerator(name="employee", sequenceName="employee_seq", allocationSize=1)
	private Integer id;
	private String username;
	private String password;
	private String email;
	@Column(name = "fname")
	private String first;
	@Column(name = "lname")
	private String lname;
	
	private Employee manager;
}

File: frontend/ERS/src/app/shared/employee.ts
import { Manager } from './manager';

export class Employee extends Manager {
    public manager: Manager;
    constructor(id: number, 
                username: string, 
                password: string, 
                email: string, 
                first: string, 
                last: string,
                manager: Manager) {
        super(id, username, password, email, first, last);
        this.manager = manager;
    }
}
